net_handler/tcp_conn.c

```diff
@@ -61,11 +61,11 @@ static void send_log_msg (int conn_fd, FILE *log_file)
 			log_msg.payload[log_msg.n_payload] = malloc(strlen(nextline) + 1);
 			strcpy(log_msg.payload[log_msg.n_payload], nextline);
 			log_msg.n_payload++;
-		}
-		else if (errno == EAGAIN) { // EAGAIN occurs when nothing to read
+		} else if (feof(log_file) != 0) { // All write ends of FIFO are closed, don't send any more logs
+			return;
```

Comment:
return only if `log_msg.n_payload == 0` otherwise break to send those messages

Reply:
👌